packages/core/src/prompts/types.ts

```diff
@@ -0,0 +1,143 @@
+import type { GetPromptResult } from '@modelcontextprotocol/sdk/types.js';
+
+/**
+ * MCP-compliant prompt argument definition
+ */
+export interface PromptArgument {
+    name: string;
+    description?: string;
+    // Optional to accommodate providers that omit it; treat as false when missing
+    required?: boolean;
+}
+
+/**
+ * MCP-compliant prompt definition
+ */
+export interface PromptDefinition {
+    name: string;
+    title?: string | undefined;
+    description?: string | undefined;
+    arguments?: PromptArgument[] | undefined;
+}
+
+/**
+ * Content types for prompt messages as per MCP spec
+ */
+export type PromptContentType = 'text' | 'image' | 'audio' | 'resource';
+
+/**
+ * Base content interface for prompt messages
+ */
+export interface PromptContent {
+    type: PromptContentType;
+}
+
+/**
+ * Text content for prompt messages
+ */
+export interface TextContent extends PromptContent {
+    type: 'text';
+    text: string;
+}
+
+/**
+ * Image content for prompt messages
+ */
+export interface ImageContent extends PromptContent {
+    type: 'image';
+    data: string; // base64-encoded image data
+    mimeType: string;
+}
+
+/**
+ * Audio content for prompt messages
+ */
+export interface AudioContent extends PromptContent {
+    type: 'audio';
+    data: string; // base64-encoded audio data
+    mimeType: string;
+}
+
+/**
+ * Embedded resource content for prompt messages
+ */
+export interface ResourceContent extends PromptContent {
+    type: 'resource';
+    resource: {
+        uri: string;
+        name: string;
+        title?: string;
+        mimeType: string;
+        text?: string;
+        data?: string; // base64-encoded blob data
+    };
+}
+
+/**
+ * Union type for all content types
+ */
+export type PromptMessageContent = TextContent | ImageContent | AudioContent | ResourceContent;
+
```

Comment:
a lot of these seem like they might be duplicates of the types we use for storing messages internally

currently our internal message types aren't proper though so we can add a TODO to re-check thse types later against the internal message types